lib/bolt/plugin/terraform.rb

```diff
@@ -24,29 +24,20 @@ def warn_missing_property(name, property)
       def lookup_targets(opts)
         state = load_statefile(opts)
 
-        resources = state.fetch('modules', {}).flat_map do |mod|
-          mod.fetch('resources', {}).map do |name, resource|
-            [name, resource.dig('primary', 'attributes')]
-          end
-        end
+        resources = extract_resources(state)
 
         regex = Regexp.new(opts['resource_type'])
 
         resources.select do |name, _resource|
           name.match?(regex)
         end.map do |name, resource|
-          unless resource.key?(opts['uri'])
-            warn_missing_property(name, opts['uri'])
-            next
-          end
+          uri = lookup(name, resource, opts['uri'])
+          next unless uri
```

Comment:
I dont think this was introduced with this work, but would always setting a `uri` here result in overriding potential config like `host` for ssh config? So specifically If I had 
```yaml
version: 2
groups:
  - name: terraform-nodes
    groups:
      - name: terraform-group-1
        target-lookups:
          - plugin: terraform
            dir: /home/cas/working_dir/BOLT-1265-tf-plugin
            resource_type: aws_instance.xenial.[0-1]
            name: id
            config:
              ssh:
                host: public_ip
```
I would get the following stack trace
```
cas@cas-ThinkPad-T460p:~/working_dir/bolt$ bolt command run whoami -t i-09ebed1560c80c46e
bundler: failed to load command: bolt (/home/cas/working_dir/bolt/.bundle/ruby/2.5.0/bin/bolt)
NoMethodError: undefined method `split' for nil:NilClass
  /home/cas/working_dir/bolt/lib/bolt/plugin/terraform.rb:88:in `lookup'
  /home/cas/working_dir/bolt/lib/bolt/plugin/terraform.rb:34:in `block in lookup_targets'
```
With the following implementation of `lookup_targets` I can set the `name` only and reference it by id
```ruby
      def lookup_targets(opts)
        state = load_statefile(opts)

        resources = extract_resources(state)

        regex = Regexp.new(opts['resource_type'])

        resources.select do |name, _resource|
          name.match?(regex)
        end.map do |name, resource|
          target = if opts.key?('name')
              real_name = lookup(name, resource, opts['name'])
              { 'name' => real_name } if real_name
            else
              uri = lookup(name, resource, opts['uri'])
              { 'uri' => uri } if uri
            end

          next unless target

          if opts.key?('config')
            target['config'] = resolve_config(name, resource, opts['config'])
          end
          target
        end.compact
      end
```
```
cas@cas-ThinkPad-T460p:~/working_dir/bolt$ bolt command run whoami -t i-09ebed1560c80c46e
Started on 34.222.221.208...
Finished on 34.222.221.208:
  STDOUT:
    ubuntu
Successful on 1 node: i-09ebed1560c80c46e
Ran on 1 node in 0.52 seconds
```

Reply:
Good call. I will update this to allow just name.